src/hdf5/read_cloudsat.py

```diff
@@ -0,0 +1,209 @@
+    invalid = np.logical_or(data < valid_min, data > valid_max)
+    dataf = data.astype(float)
+    dataf[invalid] = np.nan
+
+    # Apply scale factor according to [1].
+    dataf = dataf / scale_factor
+
+    xid = vs.find('start_time')
+    start_time = np.array(vs.attach(xid)[:]).flatten()
+    start_time = datetime.datetime.strptime(start_time[0], '%Y%m%d%H%M%S')
+
+    xid = vs.find('Latitude')
+    latid = vs.attach(xid)
+    latid.setfields('Latitude')
+    nrecs, _, _, _, _ = latid.inquire()
+    latitude = np.array(latid.read(nRec=nrecs))
+    latid.detach()
+
+    lonid = vs.attach(vs.find('Longitude'))
+    lonid.setfields('Longitude')
+    nrecs, _, _, _, _ = lonid.inquire()
+    longitude = np.array(lonid.read(nRec=nrecs))
+    lonid.detach()
+
+    demid = vs.attach(vs.find('DEM_elevation'))
+    demid.setfields('DEM_elevation')
+    nrecs, _, _, _, _ = demid.inquire()
+    dem = np.array(demid.read(nRec=nrecs))
+    demid.detach()
+
+    sfcid = vs.attach(vs.find('SurfaceHeightBin_fraction'))
+    sfcid.setfields('SurfaceHeightBin_fraction')
+    nrecs, _, _, _, _ = sfcid.inquire()
+    SurfaceHeightBin_fraction = np.array(sfcid.read(nRec=nrecs))
+    sfcid.detach()
+
+    sfcid = vs.attach(vs.find('SurfaceHeightBin'))
+    sfcid.setfields('SurfaceHeightBin')
+    nrecs, _, _, _, _ = sfcid.inquire()
+    SurfaceHeightBin = np.array(sfcid.read(nRec=nrecs))
+    sfcid.detach()
+
+    RangeBinSize = 250
+    SurfaceHeightBin_fraction[SurfaceHeightBin_fraction < -5] = 0
+    surface_height = np.zeros(height.shape[0])
+    for i, shb in enumerate(SurfaceHeightBin.flatten()):
+        surface_height[i] = height[i, shb] + RangeBinSize * SurfaceHeightBin_fraction[i]
+
+    timeid = vs.attach(vs.find('Profile_time'))
+    timeid.setfields('Profile_time')
+    nrecs, _, _, _, _ = timeid.inquire()
+    time = timeid.read(nRec=nrecs)
+    epoch_time = np.zeros(len(time))
+    for i, ss in enumerate(time):
+        time1 = start_time + datetime.timedelta(seconds=ss[0])
+        epoch_time[i] = round((time1 - epoch).total_seconds())
+
+    '''
+    units_t =  timeid.attr('units').get()
+    longname_t = timeid.attr('long_name').get()
+    '''
+    timeid.detach()
+
+    # interpolate to common height
+    '''
+    dataf1 = np.ma.array(dataf, mask=np.isnan(dataf))
+    for n in np.arange(dataf1.shape[0]):
+        dataf1[n,:] = np.interp(height[0,:], height[n,:], dataf1[n,:])
+    '''
+
+    # build output dictionary
+    csdata = {}
+    csdata['lat'] = np.squeeze(latitude)
+    csdata['lon'] = np.squeeze(longitude)
+    csdata['height'] = np.squeeze(height)
+    csdata['dem_elevation'] = np.squeeze(dem)
+    csdata['surface_height'] = np.squeeze(surface_height)
+    csdata['SurfaceHeightBin'] = np.squeeze(SurfaceHeightBin)
+    csdata['epoch_time'] = np.squeeze(epoch_time)
+    csdata['obs'] = np.squeeze(dataf)
+    csdata['zenith_angle'] = csdata['lat'].copy() * 0
+    csdata['azimuth_angle'] = csdata['lat'].copy() * 0
```

Comment:
@imoradi am I correct that this looks like we are not getting a `zenith` or `azimuth` angle for the observations?